OpenRA.Game/Traits/Player/PlayerResources.cs

```diff
@@ -111,9 +121,19 @@ public void GiveCash(int num)
 			}
 		}
 
-		public bool TakeCash(int num)
+		public bool TakeCash(int num, bool notifyLowFunds = false)
 		{
-			if (Cash + Resources < num) return false;
+			if (Cash + Resources < num)
+			{
+				if (notifyLowFunds && !string.IsNullOrEmpty(info.InsufficientFundsNotification) &&
+					owner.World.WorldTick - lastNotificationTick >= info.InsufficientFundsNotificationDelay)
+				{
+					lastNotificationTick = owner.World.WorldTick;
+					Game.RunAfterTick(() => Game.Sound.PlayNotification(owner.World.Map.Rules, owner, "Speech", info.InsufficientFundsNotification, owner.Faction.InternalName));
```

Comment:
No clue, I was wondering as well. I just took it over from the original code. I'll remove it.